File: src/main/java/net/igneo/icv/client/animation/EnchantAnimationPlayer.java
package net.igneo.icv.client.animation;

import dev.kosmx.playerAnim.api.firstPerson.FirstPersonMode;
import dev.kosmx.playerAnim.api.layered.KeyframeAnimationPlayer;
import dev.kosmx.playerAnim.core.data.KeyframeAnimation;
import org.jetbrains.annotations.NotNull;

public class EnchantAnimationPlayer extends KeyframeAnimationPlayer {
    public EnchantAnimationPlayer(KeyframeAnimation emote, int t, boolean mutable) {
        super(emote, t, mutable);
    }

    public EnchantAnimationPlayer(KeyframeAnimation emote) {
        super(emote, 0, false);
    }

    public boolean isWindingDown(float tickDelta) {
        int windDownStart = getData().endTick + ((getData().stopTick - getData().endTick) / 4);
        return ((getTick() + tickDelta) > (windDownStart + 0.5F)); // + 0.5 for smoother transition
    }

    @Override
    public @NotNull FirstPersonMode getFirstPersonMode(float tickDelta) {
        return FirstPersonMode.THIRD_PERSON_MODEL;
    }
}

File: src/main/java/net/igneo/icv/client/animation/EnchantAnimationPlayer.kt
package net.igneo.icv.client.animation

import dev.kosmx.playerAnim.api.firstPerson.FirstPersonMode
import dev.kosmx.playerAnim.api.layered.KeyframeAnimationPlayer
import dev.kosmx.playerAnim.core.data.KeyframeAnimation

class EnchantAnimationPlayer : KeyframeAnimationPlayer {

    constructor(animation: KeyframeAnimation, tick: Int, mutable: Boolean) : super(animation, tick, mutable)
    constructor(animation: KeyframeAnimation) : super(animation, 0, false)

    fun isWindingDown(tickDelta: Float): Boolean {
        val windDownStart = data.endTick + (data.stopTick - data.endTick) / 4
        return (tick + tickDelta) > (windDownStart + 0.5f)
    }

    override fun getFirstPersonMode(tickDelta: Float): FirstPersonMode = FirstPersonMode.THIRD_PERSON_MODEL
}